frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/WhenExpressionDecorator.tsx

```diff
@@ -1,35 +1,47 @@
 import * as React from 'react';
 import { Tooltip } from '@patternfly/react-core';
+import { chart_color_green_400 as successColor } from '@patternfly/react-tokens/dist/js/chart_color_green_400';
+import { global_BackgroundColor_200 as greyBackgroundColor } from '@patternfly/react-tokens/dist/js/global_BackgroundColor_200';
+import { global_BackgroundColor_light_100 as lightBackgroundColor } from '@patternfly/react-tokens/dist/js/global_BackgroundColor_light_100';
 import { global_BorderColor_100 as lightBorderColor } from '@patternfly/react-tokens/dist/js/global_BorderColor_100';
 import { useTranslation } from 'react-i18next';
-import { runStatus } from '../../../utils/pipeline-augment';
+import { getRunStatusColor, runStatus } from '../../../utils/pipeline-augment';
 import { NODE_HEIGHT } from './const';
 
 import './WhenExpressionDecorator.scss';
 
 type WhenExpressionDecoratorProps = {
   width: number;
   height: number;
-  color: string;
+  color?: string;
   leftOffset?: number;
   stroke?: string;
-  status?: string;
+  status: string;
   appendLine?: boolean;
   enableTooltip?: boolean;
+  isFinallyTask: boolean;
+  isPipelineRun: boolean;
 };
 
 const WhenExpressionDecorator: React.FC<WhenExpressionDecoratorProps> = ({
   width,
   height,
-  color,
   enableTooltip,
-  stroke = lightBorderColor.value,
   appendLine = false,
   status,
   leftOffset = 2,
+  isFinallyTask,
+  isPipelineRun,
 }) => {
   const { t } = useTranslation();
   const rotation = 45; // 45deg
+  const diamondColor = isPipelineRun
```

Comment:
Can we move the colour and tooltip logic into it's own utility... something like:

```
const getWhenDiamondState = (...): [string, string];
// Essentially take whatever we need to calculate, return `[color, tooltipText]`.
```
This is a testable function, a unifies the logic in one spot. We destructure in this component and just use them as-is. Would that not make a more cleaner approach to this problem? 

I'm also not feeling this ternary inside a ternary wrapped in a ternary is a good way to convey logic. Perhaps convert it to if-statements when you put it into a utility.